File: Data-Structures/Arrays/arrays.py
# Student Grades Calculator in Python

# Taking input for number of students
num_students = int(input("Enter the number of students: "))
grades = []

# Taking input for grades
for i in range(num_students):
    grade = float(input(f"Enter grade for student {i+1}: "))
    grades.append(grade)

# Calculating statistics
average = sum(grades) / num_students
highest = max(grades)
lowest = min(grades)

# Displaying results
print(f"\nAverage Grade: {average:.2f}")
print(f"Highest Grade: {highest:.2f}")
print(f"Lowest Grade: {lowest:.2f}")

File: Data-Structures/Arrays/arrays.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Taking input for number of students
rl.question("Enter the number of students: ", (numStudents) => {
  numStudents = parseInt(numStudents);
  let grades = [];
  let count = 0;

  // Function to take grades input
  const takeInput = () => {
    if (count < numStudents) {
      rl.question(`Enter grade for student ${count + 1}: `, (grade) => {
        grades.push(parseFloat(grade));
        count++;
        takeInput();
      });
    } else {
      // Calculating statistics
      let sum = grades.reduce((a, b) => a + b, 0);
      let highest = Math.max(...grades);
      let lowest = Math.min(...grades);
      let average = sum / numStudents;

      // Displaying results
      console.log(`\nAverage Grade: ${average.toFixed(2)}`);
      console.log(`Highest Grade: ${highest.toFixed(2)}`);
      console.log(`Lowest Grade: ${lowest.toFixed(2)}`);

      rl.close();
    }
  };

  takeInput();
});
